### PR TITLE
Functional button components with css module

### PR DESCRIPTION
**Pull request by bgptr**
This diff refactors all components in `app/components/buttons/` using hooks and CSS modules.
It ditches `TextToggle.less` which is related to #3259

Also, it includes automated tests.
Part of #2438 & #2439.

### CODE REVIEW

File: app/components/buttons/KeyBlueButton/KeyBlueButton.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Button from "../Button/Button";
+import { classNames } from "pi-ui";
+import styles from "./KeyBlueButton.module.css";
+
+const KeyBlueButton = (props) => (
+  <Button
+    {...props}
+    className={classNames(
+      styles.keyBlueButton,
+      props.className && props.className
```
Comment (amassarwi) [inline app/components/buttons/KeyBlueButton/KeyBlueButton.jsx:10]: Should be enough:
```suggestion
     props.className
```

File: app/components/buttons/ModalButton.js [hunk 2]
```diff
@@ -14,75 +16,62 @@ const defaultButton = ({
...
+    onClick
+  } = props;
+  const [show, setShow] = useState(!!showModal);
 
-  hideModal() {
-    this.setState({ show: false });
-  }
+  const onShowModal = () => {
+    setShow(true);
+    onShow?.();
```
Comment (amassarwi) [inline app/components/buttons/ModalButton.js:35]: 😻

File: app/components/buttons/TextToggle/Toggle.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import styles from "./TextToggle.module.css";
+import { classNames } from "pi-ui";
+
+const Toggle = ({ leftText, rightText, activeButton, onClick }) => (
+  <div className={classNames(styles.textToggle, styles.isRow)}>
+    <div
+      className={classNames(
+        styles.textToggleButtonLeft,
+        activeButton === "left" ? styles.textToggleButtonActive : ""
```
Comment (amassarwi) [inline app/components/buttons/TextToggle/Toggle.jsx:9]:
```suggestion
        activeButton === "left" && styles.textToggleButtonActive
```

File: app/components/buttons/TextToggle/Toggle.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      )}
+      onClick={activeButton == "right" ? () => onClick("left") : null}>
+      {leftText}
+    </div>
+    <div
+      className={classNames(
+        styles.textToggleButtonRight,
+        activeButton === "right" ? styles.textToggleButtonActive : ""
```
Comment (amassarwi) [inline app/components/buttons/TextToggle/Toggle.jsx:17]:
```suggestion
        activeButton === "right" && styles.textToggleButtonActive
```

**Review by amassarwi — APPROVED** [review]
Nice work, left some minor observations.

File: app/components/buttons/KeyBlueButton/KeyBlueButton.jsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import Button from "../Button/Button";
+import { classNames } from "pi-ui";
+import styles from "./KeyBlueButton.module.css";
+
+const KeyBlueButton = (props) => (
```
Comment (amassarwi) [inline app/components/buttons/KeyBlueButton/KeyBlueButton.jsx:5]:
```suggestion
const KeyBlueButton = ({className, ...props}) => (
```

**Comment by bgptr** [comment]
@amassarwi  Thank you for your review. I've implemented your suggested changes.

File: app/components/buttons/MixerSettingsIconButton/MixerSettingsIconButton.jsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { MixerSettingsModal } from "modals";
 import ModalButton from "../ModalButton";
-import InvisibleButton from "../InvisibleButton";
+import InvisibleButton from "../InvisibleButton/InvisibleButton";
```
Comment (amassarwi) [inline app/components/buttons/MixerSettingsIconButton/MixerSettingsIconButton.jsx:3]: could add an index.js to avoid this change

**Review by amassarwi — COMMENTED** [review]
one last bit - looks like `components/buttons/ModalButton.js` should be renamed to `ModalButton.jsx`, please make sure all other components have `.jsx` as well.

File: app/components/buttons/DangerButton/DangerButton.jsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import Button from "../Button/Button";
+import { classNames } from "pi-ui";
+import styles from "./DangerButton.module.css";
+
+const DangerButton = (props) => (
```
Comment (amassarwi) [inline app/components/buttons/DangerButton/DangerButton.jsx:5]: destruct props

File: app/components/buttons/InvisibleButton/InvisibleButton.jsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import Button from "../Button/Button";
+import { classNames } from "pi-ui";
+import styles from "./InvisibleButton.module.css";
+
+const InvisibleButton = (props) => (
```
Comment (amassarwi) [inline app/components/buttons/InvisibleButton/InvisibleButton.jsx:5]: destruct props

File: app/components/buttons/ModalButton.js [hunk 9]
```diff
@@ -14,75 +16,62 @@ const defaultButton = ({
...
-  constructor(props) {
-    super(props);
-    this.state = { show: false };
-  }
-
-  showModal() {
-    this.setState({ show: true });
-    this.onShow();
-  }
+const ModalButton = (props) => {
```
Comment (amassarwi) [inline app/components/buttons/ModalButton.js:19]: desctruct props

File: app/components/buttons/SlateGrayButton/SlateGrayButton.jsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import Button from "../Button/Button";
+import { classNames } from "pi-ui";
+import styles from "./SlateGrayButton.module.css";
+
+const SlateGrayButton = (props) => (
```
Comment (amassarwi) [inline app/components/buttons/SlateGrayButton/SlateGrayButton.jsx:5]: destruct props

**Comment by bgptr** [comment]
@amassarwi Thank you for your review.

**Review by amassarwi — APPROVED** [review]

**Review by alexlyp — APPROVED** [review]

**Pull request merged**